### PR TITLE
Append to the output file instead of write

### PR DESCRIPTION
**Pull request by brummett**
This way, the output file will persist across invocations of the app, letting it record properly when used as a CGI

### CODE REVIEW

**Comment by hoelzro** [comment]
The problem with this change is that if I'm using the recorder to record a session for playback, I have to remember to remove the recording file every time I want to record a new session.  Otherwise, you'll end up with sessions strung together.

**Comment by brummett** [comment]
True, but stringing the sessions together is exactly what you want if it's a CGI.

What about using mode 'a' if ```psgi.run_once``` is set in the PSGI environment, and 'w' otherwise?

**Comment by hoelzro** [comment]
@brummett That works for me!

**Comment by brummett** [comment]
Added more commits to open the output file in append mode only if ```psgi.run_once``` is true.

**Comment by hoelzro** [comment]
@brummett Looks good to me, thanks!

File: lib/Plack/Middleware/Recorder.pm [hunk 1]
```diff
@@ -22,13 +22,26 @@ sub prepare_app {
     $self->start_url('/recorder/start') unless defined $self->start_url;
     $self->stop_url('/recorder/stop')   unless defined $self->stop_url;
 
-    my $output = $self->{'output'};
+    my $output = delete $self->{output};
     croak "output parameter required" unless defined $output;
 
-    $output = $self->{'output'} = IO::File->new($output, 'w') or croak $!
-        unless ref $output;
+    if (ref $output) {
```
Comment (hoelzro) [inline lib/Plack/Middleware/Recorder.pm]: @brummett I was looking at the commit that introduced this, and I'm not sure why you lazily initialize the output file handle.  Does doing it here interfere with CGI somehow?

Comment (brummett) [inline lib/Plack/Middleware/Recorder.pm]: There's no way to tell what the value for the ```psgi.run_once``` flag is during prepare_app().  It's not until we're handling the first request that we get passed the PSGI environment and can look at the flag.

Comment (hoelzro) [inline lib/Plack/Middleware/Recorder.pm]: Ah, good point!

**Pull request merged**